File: c#/1603. Design Parking System/Solution.cs
public class ParkingSystem {
    private int big;
    private int medium;
    private int small;
    public ParkingSystem(int big, int medium, int small) {
        this.big = big;
        this.medium = medium;
        this.small = small;
    }
    
    public bool AddCar(int carType) {
        switch(carType)
        {
            case 1:
                if (big > 0)
                {
                    big--;
                    return true;
                }
                return false;
            case 2:
                if (medium > 0)
                {
                    medium--;
                    return true;
                }
                return false;
            case 3:
                if (small > 0)
                {
                    small--;
                    return true;
                }
                return false;
            default:
                return false;
        }
    }
}

File: python/1603. Design Parking System/Solution.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.spots = [big, medium, small]

    def addCar(self, carType: int) -> bool:
        self.spots[carType - 1] -= 1
        return self.spots[carType - 1] >= 0